### PR TITLE
Issue85 metadata api

### LINKED ISSUE #85: Update erc1155 connector to not require ABI published in ethconnect
Both this connector and the [erc20-erc721](https://github.com/hyperledger/firefly-tokens-erc20-erc721) connector were designed to work against [firefly-ethconnect](https://github.com/hyperledger/firefly-ethconnect). However, ethconnect supports two modes of operation:
1. you upload ABI information to ethconnect, and it generates and hosts a "friendly" REST-like API instance for invoking and listening to the associated contract
2. you invoke the raw RPC wrappers on ethconnect and provide ABI metadata inline on each call

Currently this connector is still using (1), while erc20-erc721 is using (2). The brand new [firefly-evmconnect](https://github.com/hyperledger/firefly-evmconnect) supports only (2). Therefore, for maximum flexibility, this connector should also be converted to (2).

Suggested steps:
1. check in the ABI JSON for the well-defined ERC1155 [sample contract](https://github.com/hyperledger/firefly-tokens-erc1155/blob/main/samples/solidity/contracts/ERC1155MixedFungible.sol) in this repository, and import bits of this JSON as [done in erc20-erc721](https://github.com/hyperledger/firefly-tokens-erc20-erc721/blob/465bf5fd6974f34a53ef3cd0b05c2364ad516663/src/tokens/tokens.service.ts#L30)
2. switch all calls that invoke transactions, perform queries, or create subscriptions against ethconnect to use the top-level APIs as [used](https://github.com/hyperledger/firefly-tokens-erc20-erc721/blob/465bf5fd6974f34a53ef3cd0b05c2364ad516663/src/tokens/tokens.service.ts#L434) [by](https://github.com/hyperledger/firefly-tokens-erc20-erc721/blob/465bf5fd6974f34a53ef3cd0b05c2364ad516663/src/tokens/tokens.service.ts#L414) [erc20-erc721](https://github.com/hyperledger/firefly-tokens-erc20-erc721/blob/465bf5fd6974f34a53ef3cd0b05c2364ad516663/src/event-stream/event-stream.service.ts#L255)
3. alter the [environment config](https://github.com/hyperledger/firefly-tokens-erc1155/blob/958aff36a62ffd96c29c0dabf554243c6368a648/.env#L3) to accept a `CONTRACT_ADDRESS` instead of a friendly ethconnect ABI instance - noting that we can provide a migration path to derive a contract address from an instance path [as we had to do in core](https://github.com/hyperledger/firefly/blob/bd63308c8cc855b4859ffc114d4ef6fcdcd7f42e/internal/blockchain/ethereum/ethereum.go#L833) at one point too

This should all be completed using ethconnect. Once it is verified working there, we can move on to testing against the new evmconnect.

### PR DESCRIPTION
**Pull request by Chengxuan**
Resolves #85 

- Checked in ABI json from ERC1155
- updated token service to use the metadata driven API
- introduced a new env var CONTRACT_ADDRESS with migration logic if it's not set

### CODE REVIEW

**Review by awrichar — COMMENTED** [review]
Reviewing this now - two style/formatting notes:
* Can you run `npm run format`?
* Not sure why the entire samples/solidity/package-lock.json file was rewritten - looks like a bunch of whitespace changed for some reason. Would be nice to not overwrite it if it's not needed.

File: src/tokens/tokens.service.ts [hunk 1]
```diff
@@ -141,13 +149,26 @@ export class TokensService {
     );
 
     this.supportsCustomUri = await this.queryUriSupport();
+
+    if (!this.contractAddress) {
```
Comment (awrichar) [inline src/tokens/tokens.service.ts:153]: `init()` is not guaranteed to be called on every start (only has to be called once ever). So we need to be sure `this.contractAddress` is initialized somewhere else. Might want to do it lazily the first time a query/transaction happens?

Comment (awrichar) [reply src/tokens/tokens.service.ts:153]: Come to think of it, there's probably an existing bug with the initialization of `this.supportsCustomUri` above as well.

**Review by awrichar — CHANGES_REQUESTED** [review]
Just one functional change needed surrounding init. Otherwise looks great!

**Comment by Chengxuan** [comment]
@awrichar thank you for the comments. I've addressed them. Ready to be reviewed again.

File: src/tokens/tokens.service.ts [hunk 2]
```diff
@@ -147,45 +147,53 @@ export class TokensService {
...
       const response = await this.wrapError(
         lastValueFrom(
           this.http.get<ContractInfoResponse>(`${this.instanceUrl}`, {
             ...basicAuth(this.username, this.password),
           }),
         ),
       );
-      this.contractAddress = response.data.address;
+      this.contractAddress = response.data.address.toLowerCase();
+      this.logger.debug(`s: ${this.contractAddress}`);
```
Comment (awrichar) [inline src/tokens/tokens.service.ts:165]: Could annotate this log a little better (or remove if it's not needed now)

**Review by awrichar — APPROVED** [review]

**Review by Chengxuan — COMMENTED** [review]
debug log remove in https://github.com/hyperledger/firefly-tokens-erc1155/pull/91

**Pull request merged**